### PR TITLE
@craigspaeth Purchase test

### PR DESCRIPTION
**Pull request by 1aurabrown**
A few things:

* There is a distinction between an artwork being _eligible_ for the purchase flow and the outcome of the split test. If a user is in the test group but looking at an e-commerce work, we change the copy of the button to 'Purchase' even though this artwork is not eligible for the test itself. @anipetrov let me know if we want to 'view' the experiment in this case as well.

* Right now I am only notifying segment that the 'experiment' is being viewed if the artwork is eligible for the purchase flow -- that means it has a price, not a range, and is not e-commerce. Other artworks (e-commerce, no price, not for sale) do not trigger the event.

* Not sure if there is a better way to inject a heading into the 'minimal nav bar'. let me know if you think there is.

* Needs error handling

* Needs responsive version

* Need to determine wether or not we will require users to log in, pending forced log in test results.

* Needs platform work to get get data flag indicating whether price is a range or not--right now i'm doing string matching.

* Kind of gross to share a helper across 2 apps (you'll see the `artwork_purchase` helper to determine if a work is eligible is being shared with the `artwork` app). This is not a component though, so not sure how we want file hierarchy to be.

* success page is just a re-render with a different template and Backbone history update a url called `#{artwork_id}/thank-you`. Right now it does a replace, we should clarify if we want to make a history entry or not. Hitting back after success takes you back to the artwork. Previously I set replace: false, but it does not hard-reload the checkout form when you hit back from the success page, it just changes the url and nothing else.

* Purchase page will redirect if you are not in the test group. If you are in the test group, it will redirect after fetching the artwork if the artwork is not eligible for the purchase flow.

* Hard load of Thank You page redirects to checkout page.

### CODE REVIEW

**Comment by 1aurabrown** [comment]
If I am in the test group artworks appear like this:

e commerce - purchase button leads to order flow:
screen shot 2016-11-02 at 3 50 40 pm

inquireable with a range:
screen shot 2016-11-02 at 3 52 44 pm

eligible for purchase flow:
screen shot 2016-11-02 at 3 52 56 pm

in the last case, hitting the purchase button takes you here:
screen shot 2016-11-01 at 4 39 32 pm

(yellow from autocomplete)

right now the inquiries submit without error, i have a placeholder success page.

**Comment by craigspaeth** [comment]
Thanks for the thorough PR! Looking forward to digging in and picking up later.

**Comment by 1aurabrown** [comment]
i will fix the failing test! it has to do with how the template is rendered on the server vs how it's rendered with the `render` method in tests.

File: apps/artwork/components/commercial/index.jade [hunk 1]
```diff
@@ -1,6 +1,11 @@
 unless artwork.is_in_auction || artwork.is_for_sale == false
-  form.artwork-commercial( class='js-artwork-commercial' )
-    include templates/index
-    | Questions about buying art on Artsy?
-    | Read our&nbsp;
-    a.collector-faq( href='/#' ) Collector FAQ
\ No newline at end of file
+  .artwork-commercial( class='js-artwork-commercial' )
+    form.artwork-commercial__form
+      if purchaseFlow && eligibleForPurchase
```
Comment (craigspaeth) [inline apps/artwork/components/commercial/index.jade]: 👌

File: apps/artwork/components/commercial/view.coffee [hunk 2]
```diff
@@ -16,29 +18,58 @@ module.exports = class ArtworkCommercialView extends Backbone.View
...
+    'click .js-artwork-inquire-button'  : 'inquire'
+    'click .js-artwork-acquire-button'  : 'acquire'
+    'click .collector-faq'              : 'openCollectorModal'
 
   initialize: ({ @data }) ->
     { artwork } = @data
-    { @fair } = artwork
+
+    if isEligible artwork
+      splitTest('purchase_flow').view()
```
Comment (craigspaeth) [inline apps/artwork/components/commercial/view.coffee]: 👍

File: apps/artwork/components/commercial/view.coffee [hunk 3]
```diff
@@ -16,29 +18,58 @@ module.exports = class ArtworkCommercialView extends Backbone.View
...
+    @user = User.instantiate()
+    @artwork.fetch().then =>
+      @artwork.related().fairs.add @data.artwork.fair
+      @modal = openInquiryQuestionnaireFor
+        user: @user
+        artwork: @artwork
+        inquiry: @inquiry
+
+      # Success
+      @listenToOnce @inquiry, 'sync', =>
```
Comment (craigspaeth) [inline apps/artwork/components/commercial/view.coffee]: 👍

File: apps/artwork_purchase/client/index.coffee [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    @$el.html successTemplate { @artwork }
+
+class ThankYouView extends Backbone.View
+
+module.exports.init = ->
+  user = CurrentUser.orNull()
+  artwork = ARTWORK
+  Backbone.history.start pushState: true
+  new PurchaseForm { user, artwork, el: $('#purchase-page') }
+  new ThankYouView { user, artwork, el: $('#purchase-page') }
```
Comment (craigspaeth) [inline apps/artwork_purchase/client/index.coffee]: View's are looking good so far 👍

File: apps/artwork_purchase/helpers.coffee [hunk 5]
```diff
@@ -0,0 +1,49 @@
+_ = require 'underscore'
+
+module.exports =
```
Comment (craigspaeth) [inline apps/artwork_purchase/helpers.coffee]: 👍  nice extracting helpers here.

File: components/mixins/form.coffee [hunk 6]
```diff
@@ -62,7 +62,7 @@ module.exports =
     form = @serializeInputs $form
     _.extend form, booleans
 
-  # Checks for required fileds then sets the data-state to error
+  # Checks for required fields then sets the data-state to error
```
Comment (craigspaeth) [inline components/mixins/form.coffee]: 👀

**Comment by craigspaeth** [comment]
Code is looking go so far. You've got a solid foundation to work from here! 👏  good job!

**Comment by 1aurabrown** [comment]
hm... i rebased and now all the intermediate commits show up here

**Pull request closed without merging**